[Bug 619799] Different error message for files exceeding maxsize

pollti opened the pull request:
This should throw another error message if a file is too large.
See [bug 619799](https://bugzilla.mozilla.org/show_bug.cgi?id=619799) for further information.

mythmon commented on kitsune/sumo/static/js/gallery.js:
```diff
@@ -203,11 +208,15 @@
-                    file.size < CONSTANTS.max_size[type]);
+            return (in_array(file_ext, CONSTANTS.extensions[type]));
+        },
+        isNotTooLarge: function ($input) {
```
It is odd that this is `isNotTooLarge`, and you only ever call it like `!isNotTooLarge(file)`. It is like a double negative. It would make more sense to have it be `isTooLarge()`, and call it directly. I'm also not convinced it needs to be a separate function, since it is only a couple lines and isn't re-used anywhere.

mythmon commented on kitsune/sumo/static/js/gallery.js:
```diff
@@ -203,11 +208,15 @@
         },
         /*
-         * Fired when upload starts, if isValidFile returns true
+         * Fired when upload starts, if isValidFile and isNotTooLarge return true
```
This comment shouldn't reference `isNotTooLarge`, because that function is entirely a part of `isValidFile`. Sine `startUpload` doesn't make any direct reference to `isNotTooLarge`, this is kind of confusing.

mythmon commented on kitsune/sumo/static/js/gallery.js:
```diff
@@ -256,8 +265,8 @@
-         * Fired after upload is complete, if isValidFile is true, and server
-         * returned succes.
+         * Fired after upload is complete, if isValidFile and isNotTooLarge
+         * are true, and server returned succes.
```
Same here, `isNotTooLarge` is an implementation details of `isValidFile`, not something that needs documented on this function.

mythmon commented on kitsune/sumo/static/js/gallery.js:
```diff
@@ -309,7 +318,7 @@
         },
         /*
-         * Fired if isValidFile is false or server returned failure.
+         * Fired if isValidFile or isNotTooLarge is false or server returned failure.
```
Same here, `isNotTooLarge` is an implementation details of `isValidFile`, not something that needs documented on this function.

mythmon commented:
Thanks for the patch! This is pretty well done, despite how many comments I left. It is functional, and I just needs a bit better organization.

pollti commented:
I did the following here:
* reverted the comment changes
* deleted the function isNotTooLarge
* changed the beforeSubmit function to include the following test:
```javascript
if ($file[0].files[0].size >= CONSTANTS.max_size[$file.attr('name')]) {
   self.uploadError($file, 'toolarge');
   return false;
}
```
An alternative way to do this would be to write it the following way…
```javascript
var file2 = $file[0].files[0],
    type = $file.attr('name');
if(file2.size < CONSTANTS.max_size[type]);
```
…but we should find another name for file2 then ;-)

mythmon commented:
Gah. I'm sorry. I totally misread your original patch. This area of the code is pretty convoluted, so I messed up. I thought that your `isNotTooLarge` function was called from inside `isValidFiles()`, which is why I said it shouldn't be its own function. That's not true, so the way you had it before was better. So sorry.

If you prefer this way, we can keep it. If you'd prefer to go back to what was there before, which I'm pretty sure is actually fine except the function name is awkward, then an easy way to do that would be:

```shell
# Undo the last two changes
$ git reset --hard 3a6335c
# Update the branch. Since this is destructive, you need to force it,
$ git push --force
# now make your changes as normal
```

Again, sorry for my confusion.

mythmon commented:
Thanks!
